hazelcast/src/main/java/com/hazelcast/jet/impl/JetServiceBackend.java

```diff
@@ -379,4 +405,55 @@ public void startScanningForJobs() {
         jobCoordinationService.startScanningForJobs();
     }
 
+    // Store the metadata about the job jar that is uploaded
+    public void storeJobMetaData(JobMetaDataParameterObject parameterObject) {
+        jobUploadStore.processJobMetaData(parameterObject);
+    }
+
+    // Store a part of job jar that is uploaded
+    public JobMetaDataParameterObject storeJobMultiPart(JobMultiPartParameterObject parameterObject) {
+        JobMetaDataParameterObject result = null;
+        try {
+            result = jobUploadStore.processJobMultipart(parameterObject);
+        } catch (Exception exception) {
+            jobUploadStore.remove(parameterObject.getSessionId());
+            sneakyThrow(exception);
+        }
+        return result;
+    }
+
+    public void checkIfCanExecuteJar() {
```

Comment:
This method is called as the first thing when we receive meta data, so that if job upload is not possible, the operation can be aborted immediately . 
If we wait until execution stage, a  big jar may already be uploaded just to find that i resource upload is not enabled

Reply:
I mean to check it also in the `storeJobMultiPart`.

We have:

```
jetServiceBackend.checkIfCanExecuteJar();
jetServiceBackend.storeJobMetaData(jobMetaDataParameterObject);

jetServiceBackend.checkIfCanExecuteJar();
jetServiceBackend.executeJar(parameterObject);        
```
I am saying the `checkIfCanExecuteJar` should go to those methods in JetServiceBackend class and it can be private.

Reply:
checkIfCanExecuteJar is called from executeJar() and storeJobMetaData()